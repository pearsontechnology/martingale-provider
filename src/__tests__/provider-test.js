// provider-test.js
import React from 'react';
import PropTypes from 'prop-types';
import {Provider} from '../';
import renderer from 'react-test-renderer';

const Greet=({name='World'})=>(<div>Hello {name}!</div>);
Greet.propTypes={
  name: PropTypes.string
};

test('Renders a basic element', () => {
  const component = renderer.create(
    <Provider Component='span' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a custom component', () => {
  const component = renderer.create(
    <Provider Component={Greet} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a custom component with static props', () => {
  const component = renderer.create(
    <Provider Component={Greet} props={{name: 'Test'}}/>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a custom component with mapped values', () => {
  const mapper = ({value})=>{
    return {
      name: value
    };
  };
  const component = renderer.create(
    <Provider Component={Greet} props={{value: 'Mapped'}} mapper={mapper} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
