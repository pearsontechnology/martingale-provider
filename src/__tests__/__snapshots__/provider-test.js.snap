// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a basic element 1`] = `<span />`;

exports[`Renders a custom component 1`] = `
<div>
  Hello 
  World
  !
</div>
`;

exports[`Renders a custom component with mapped values 1`] = `
<div>
  Hello 
  Mapped
  !
</div>
`;

exports[`Renders a custom component with static props 1`] = `
<div>
  Hello 
  Test
  !
</div>
`;
